ledger/participant-integration-api/src/main/scala/platform/store/dao/CommandCompletionsTable.scala

```diff
@@ -25,19 +24,14 @@ private[platform] object CommandCompletionsTable {
   private val acceptedCommandParser: RowParser[CompletionStreamResponse] =
     sharedColumns ~ str("transaction_id") map {
       case offset ~ recordTime ~ commandId ~ transactionId =>
-        CompletionStreamResponse(
-          checkpoint = toApiCheckpoint(recordTime, offset),
-          completions = Seq(Completion(commandId, Some(Status()), transactionId)),
-        )
+        CompletionFromTransaction.acceptedCompletion(recordTime, offset, commandId, transactionId)
     }
 
   private val rejectedCommandParser: RowParser[CompletionStreamResponse] =
     sharedColumns ~ int("status_code") ~ str("status_message") map {
       case offset ~ recordTime ~ commandId ~ statusCode ~ statusMessage =>
-        CompletionStreamResponse(
-          checkpoint = toApiCheckpoint(recordTime, offset),
-          completions = Seq(Completion(commandId, Some(Status(statusCode, statusMessage)))),
-        )
+        val status = StatusProto.of(statusCode, statusMessage, Seq.empty)
+        CompletionFromTransaction.rejectedCompletion(recordTime, offset, commandId, status)
```

Comment:
Shouldn't this parser check for rejection_status?

Reply:
Not in this version, because this uses the "mutable" index, to which we're not making changes.